monkey/agent_plugins/exploiters/hadoop/src/hadoop_command_builder.py

```diff
@@ -1,71 +1,67 @@
+    IAgentCommandBuilderFactory,
+    ILinuxAgentCommandBuilder,
+    IWindowsAgentCommandBuilder,
+    LinuxDownloadMethod,
+    LinuxDownloadOptions,
+    LinuxRunOptions,
+    TargetHost,
+    WindowsDownloadMethod,
+    WindowsDownloadOptions,
+    WindowsRunOptions,
+    WindowsShell,
 )
+from monkeytypes import OperatingSystem
 
 
 def build_hadoop_command(
-    agent_id: AgentID,
     target_host: TargetHost,
-    servers: Sequence[str],
-    current_depth: int,
-    agent_destination_path: str,
+    agent_destination_path: PurePath,
     agent_download_url: str,
-    agent_otp_environment_variable: str,
-    otp: str,
+    agent_command_builder_factory: IAgentCommandBuilderFactory,
 ) -> str:
-    monkey_cmd = build_monkey_commandline(agent_id, servers, current_depth + 1)
-
     if target_host.operating_system in [
         OperatingSystem.WINDOWS,
         None,  # if unknown OS, default to Windows (based on
         # https://www.shodan.io/search/facet?query=hadoop&facet=os)
     ]:
-        base_command = HADOOP_WINDOWS_COMMAND_TEMPLATE
+        agent_destination_path = PureWindowsPath(agent_destination_path)
+        download_options: Union[
+            WindowsDownloadOptions, LinuxDownloadOptions
+        ] = WindowsDownloadOptions(
+            agent_destination_path=agent_destination_path,
+            download_method=WindowsDownloadMethod.WEB_REQUEST,
+            download_url=agent_download_url,
+        )
+
+        run_options: Union[WindowsRunOptions, LinuxRunOptions] = WindowsRunOptions(
+            agent_destination_path=agent_destination_path,
+            dropper_execution_mode=DropperExecutionMode.NONE,
+            shell=WindowsShell.POWERSHELL,
+        )
+
+        agent_command_builder: Union[
+            IWindowsAgentCommandBuilder, ILinuxAgentCommandBuilder
+        ] = agent_command_builder_factory.create_windows_agent_command_builder()
     else:
-        base_command = HADOOP_LINUX_COMMAND_TEMPLATE
+        agent_destination_path = PurePosixPath(agent_destination_path)
+        download_options = LinuxDownloadOptions(
+            agent_destination_path=agent_destination_path,
+            download_method=LinuxDownloadMethod.WGET,
+            download_url=agent_download_url,
+        )
+
+        run_options = LinuxRunOptions(
+            agent_destination_path=agent_destination_path,
+            dropper_execution_mode=DropperExecutionMode.NONE,
+        )
+
+        agent_command_builder = agent_command_builder_factory.create_linux_agent_command_builder()
 
-    return base_command % {
-        "monkey_path": agent_destination_path,
-        "http_path": agent_download_url,
-        "monkey_type": MONKEY_ARG,
-        "parameters": monkey_cmd,
-        "agent_otp_environment_variable": agent_otp_environment_variable,
-        "agent_otp": otp,
-    }
+    agent_command_builder.build_download_command(download_options)  # type: ignore
+    agent_command_builder.build_run_command(run_options)  # type: ignore
```

Comment:
Specify the specific error code being ignored instead of blindly ignoring all errors.

Example:  `# type: ignore [arg-type]`